src/fstext/pre-determinize-inl.h

```diff
@@ -547,9 +547,9 @@ void PreDeterminize(MutableFst<Arc> *fst,
     for (typename map<pair<StateId, ArcId>, size_t>::iterator m_iter = m_map.begin();
         m_iter != m_map.end();
         ++m_iter) {
-      pair<StateId, ArcId> pr = m_iter->first;
-      StateId state = pr.first;
-      ArcId arcpos = pr.second;
+      pair<StateId, ArcId> m_pr = m_iter->first;
```

Comment:
Like `StateId state = m_iter->first.first; ArcId arcpos = m_iter->first.second;`? Looks wonderfully ugly to me. Just to make sure you really really want _that_.